### PR TITLE
Modify circuit breaker

### PR DESCRIPTION
**Pull request by xianglinghui**
1. Add package flowtoken, which can control request traffic and adjust flow threshold automatically based on success rate per second.
2. Modify Redis client and Cluster client circuit breaker, change it to work with flowtoken.

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -1164,6 +1170,32 @@ func (c *ClusterClient) PoolStats() *PoolStats {
 	return &acc
 }
 
+// GetFlowtokenStates returns all flowtoken status.
+func (c *ClusterClient) GetFlowtokenStates() []*flowtoken.Snapshot {
```
Comment (vmihailenco) [inline cluster.go]: Looks like this func is unused?

**Comment by vmihailenco** [comment]
Can you please rebase this on master and fix build? Specifically please replace convey with https://github.com/onsi/ginkgo - that is what we use for testing go-redis.

File: cluster.go (hunk 1 above)
Comment (xianglinghui) [reply cluster.go]: User can trace flowtoken states by calling this func periodly.

**Comment by xianglinghui** [comment]
@vmihailenco Sorry for the late reply to you, I am on vacation these days.

I have replaced convey with ginkgo, and rebase my branch on master. When I do "rebase" for the latest code, there have some weird conflicts, so I aborted the latest rebase. Please check whether I need to do rebase again.

**Pull request closed without merging**